cdap-ui/app/cdap/components/AppOverview/index.js

```diff
@@ -0,0 +1,193 @@
+require('./AppOverview.less');
+import OverviewTabConfig from './OverviewTabConfig';
+import ConfigurableTab from 'components/ConfigurableTab';
+import ApplicationMetrics from 'components/EntityCard/ApplicationMetrics';
+import shortid from 'shortid';
+
+export default class AppOverview extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      entity: this.props.entity,
+      entityDetail: {
+        name: '',
+        artifact: {
+          version: '--',
+          name: '--'
+        },
+        programs: [],
+        datasets: [],
+        streams: []
+      },
+      dimension: {}
+    };
+  }
+  getChildContext() {
+    return {
+      entity: this.state.entityDetail
+    };
+  }
+  componentDidMount() {
+    let {right, left, width} = this.overviewRef.parentElement.getBoundingClientRect();
+    if (right && left) {
+      this.setState({
+        dimension: {
+          right,
+          left: ( left < width ? width: left)
+        }
+      });
+    }
+    let namespace = objectQuery(this.props.entity, 'metadata', 'entityId', 'id', 'namespace', 'id');
+    let appId = this.props.entity.id;
+    MyAppApi
+      .get({
+        namespace,
+        appId
+      })
+      .map(entityDetail => {
+        let programs = entityDetail.programs.map(prog => {
+          prog.uniqueId = shortid.generate();
+          return prog;
+        });
+        let datasets = entityDetail.datasets.map(dataset => {
+          dataset.entityId = {
+            id: {
+              instanceId: dataset.name
+            },
+            type: 'datasetinstance'
+          };
+          dataset.uniqueId = shortid.generate();
+          return dataset;
+        });
+
+        let streams = entityDetail.streams.map(stream => {
+          stream.entityId = {
+            id: {
+              streamName: stream.name
+            },
+            type: 'stream'
+          };
+          stream.uniqueId = shortid.generate();
+          return stream;
+        });
+        entityDetail.streams = streams;
+        entityDetail.datasets = datasets;
+        entityDetail.programs = programs;
+        return entityDetail;
+      })
+      .subscribe(entityDetail => {
+        this.setState({ entityDetail });
+      });
+  }
+  render() {
+    let style={};
+    let {left, right} = this.state.dimension;
+    let {right: parentRight} = this.props.parentdimension;
+    if (this.overviewRef) {
+      // FIXME: This is a hack. We need to make this as an elegant component that will intelligently
+      // figure out the position based on the intent and containing component.
+      if (this.props.position === 'right') {
+        style.left = 310;
+        style.width = `calc(100vw - ${right + 30}px)`; // factoring the container-fluid on body
+      }
+      if (this.props.position === 'left') {
+        style.right = 310;
+        style.width = `calc(100vw - ${(parentRight - left) + 30 + 20}px)`; // factoring the container-fluid on body
```

Comment:
doing the calc as inline style, will it create problem with cross-browser compatibilty?